### PR TITLE
core: optimize group deletion (delays deletion of unused contacts)

### PR DESCRIPTION
**Pull request by spaced4ndy**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -122,6 +122,7 @@ defaultChatConfig =
       testView = False,
       initialCleanupManagerDelay = 30 * 1000000, -- 30 seconds
       cleanupManagerInterval = 30 * 60, -- 30 minutes
+      cleanupManagerStepDelay = 3 * 1000000, -- 3 seconds
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:125]: why 3 seconds?

File: src/Simplex/Chat.hs [hunk 2]
```diff
@@ -837,21 +838,22 @@ processChatCommand = \case
...
       pure $ CRContactConnectionDeleted user conn
     CTGroup -> do
-      Group gInfo@GroupInfo {membership} members <- withStore $ \db -> getGroup db user chatId
+      Group gInfo@GroupInfo {membership} members <-
+        withStoreCtx (Just "APIDeleteChat CTGroup, getGroup") $ \db -> getGroup db user chatId
       let isOwner = memberRole (membership :: GroupMember) == GROwner
           canDelete = isOwner || not (memberCurrent membership)
       unless canDelete $ throwChatError $ CEGroupUserRole gInfo GROwner
-      filesInfo <- withStore' $ \db -> getGroupFileInfo db user gInfo
+      filesInfo <- withStoreCtx' (Just "APIDeleteChat CTGroup, getGroupFileInfo") $ \db -> getGroupFileInfo db user gInfo
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:846]: why do we need to use withStoreCtx everywhere?

File: src/Simplex/Chat/Migrations/M20230608_contacts_marked_for_deletion.hs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+CREATE INDEX msg_delivery_events_msg_delivery_id ON msg_delivery_events(msg_delivery_id);
+|]
+
+down_m20230608_contacts_marked_for_deletion :: Query
+down_m20230608_contacts_marked_for_deletion =
+  [sql|
+DROP INDEX msg_delivery_events_msg_delivery_id;
+
+ALTER TABLE contacts DROP COLUMN marked_for_deletion;
```
Comment (epoberezkin) [inline src/Simplex/Chat/Migrations/M20230608_contacts_marked_for_deletion.hs:21]: I think "deleted" is a conventional name here

File: src/Simplex/Chat/Types.hs [hunk 4]
```diff
@@ -176,7 +176,8 @@ data Contact = Contact
     mergedPreferences :: ContactUserPreferences,
     createdAt :: UTCTime,
     updatedAt :: UTCTime,
-    chatTs :: Maybe UTCTime
+    chatTs :: Maybe UTCTime,
+    markedForDeletion :: Bool
```
Comment (epoberezkin) [inline src/Simplex/Chat/Types.hs:180]: I don't see this field ever used - why it is added?

**Review by epoberezkin — COMMENTED** [review]
"deleted" can be removed from the type, it will reduce the noice.

Also, so queries in Contacts table don't check "deleted" column.

File: src/Simplex/Chat.hs (hunk 1 above)
Comment (spaced4ndy) [reply src/Simplex/Chat.hs:124]: arbitrary, to give other queries period when cleanup manager queries don't occupy database

File: src/Simplex/Chat.hs (hunk 2 above)
Comment (spaced4ndy) [reply src/Simplex/Chat.hs:845]: it's not necessary

**Pull request merged**